src/rust-crypto/OutgoingRequestProcessor.ts

```diff
@@ -105,7 +105,20 @@ export class OutgoingRequestProcessor {
         }
 
         if (msg.id) {
-            await this.olmMachine.markRequestAsSent(msg.id, msg.type, resp);
+            try {
+                await this.olmMachine.markRequestAsSent(msg.id, msg.type, resp);
+            } catch (e) {
+                // Ignore errors which are caused by the olmMachine having been freed. The exact error message depends
+                // on whether we are using a release or develop build of rust-sdk-crypto-wasm.
+                if (
```

Comment:
It would be good if `olmMachine` can be asked about its status and return something like "shut down". Otherwise, this may hide bugs in the implementation.

Reply:
I don't disagree, this might be preferable. Indeed, I went looking at the (generated) OlmMachine code to see if there is a nice way before doing this.  There is in fact an `OlmMachine.ptr` property which we could check, but it's not exposed via typescript.

But then: is it worth it? I think it's vanishingly unlikely we'd see either of these errors for any reason other than the OlmMachine having been freed.